markets/storageadapter/dealpublisher.go

```diff
@@ -443,3 +467,83 @@ func (p *DealPublisher) filterCancelledDeals() {
 	}
 	p.pending = filtered
 }
+
+func (p *DealPublisher) queueDeals(pdeal *pendingDeal) {
```

Comment:
I don't think we need the queueDeals method, let's just add some checks to processNewDeal